### PR TITLE
[1LP][RFR] Fixed test_alert_snmp

### PR DESCRIPTION
**Pull request by quarckster**
{{pytest: -v -k test_alert_snmp --long-running --use-provider vsphere6-nested}}

### CODE REVIEW

File: cfme/tests/control/test_alerts.py [hunk 1]
```diff
@@ -354,7 +354,7 @@ def test_alert_timeline_cpu(request, vm, set_performance_capture_threshold, prov
 
 
 @pytest.mark.provider(CANDU_PROVIDER_TYPES)
-def test_alert_snmp(request, appliance, provider, setup_snmp, setup_candu, vm, wait_candu,
+def test_alert_snmp(request, appliance, provider, setup_snmp, vm, setup_candu, wait_candu,
```
Comment (RonnyPfannschmidt) [inline cfme/tests/control/test_alerts.py]: **question** is there a sensible way to set up a provider in a way that can be shared by `vm` and `setup_candu`

having a magic dependency on order of fixtures means we are open certain kinds of human error that may be avoidable

Comment (jkrocil) [reply cfme/tests/control/test_alerts.py]: I think the right thing to do here is to add `vm` to the list of fixtures required by `setup_candu`.

Comment (quarckster) [reply cfme/tests/control/test_alerts.py]: @jkrocil vm and setup_candu fixtures have different scopes. I agree it requires refactoring, but it should be done in a separate PR.

**Comment by psav** [comment]
**REQUIRED**: If @RonnyPfannschmidt is insinuating that we may at some point get random order related failures from this, then we should add a TODO as part of this PR

**Comment by quarckster** [comment]
@psav @RonnyPfannschmidt please take a look. I tried to avoid dependence on the fixtures order.

**Pull request merged**